third_party/move/move-compiler-v2/src/pipeline/uninitialized_use_checker.rs

```diff
@@ -0,0 +1,381 @@
+        for (offset, bc) in target.get_bytecode().iter().enumerate() {
+            bc.sources().iter().for_each(|src| {
+                if let Some(state @ (Initialized::Maybe | Initialized::No)) =
+                    annotation.get_initialized_state(*src, offset as CodeOffset)
+                {
+                    target.global_env().error(
+                        &target.get_bytecode_loc(bc.get_attr_id()),
+                        &format!(
+                            "use of {}unassigned {}",
+                            match state {
+                                Initialized::No => "",
+                                Initialized::Maybe => "possibly ",
+                                Initialized::Yes => panic!("ICE: should be unreachable"),
+                            },
+                            target.get_local_name_for_error_message(*src)
+                        ),
+                    );
+                }
+            });
+        }
+    }
+
+    /// Registers initialized state annotation formatter at the given function target.
+    /// Helps with testing and debugging.
+    pub fn register_formatters(target: &FunctionTarget) {
+        target.register_annotation_formatter(Box::new(format_initialized_state_annotation));
+    }
+}
+
+impl FunctionTargetProcessor for UninitializedUseChecker {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        func_env: &FunctionEnv,
+        mut data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if func_env.is_native() {
+            // We don't have to look inside native functions.
+            return data;
+        }
+        let target = FunctionTarget::new(func_env, &data);
+        let analysis =
+            InitializedStateAnalysis::new(target.get_parameter_count(), target.get_local_count());
+        let annotation = analysis.analyze(&target);
+        self.perform_checks(&target, &annotation);
+        data.annotations.set(annotation, true); // for testing.
+        data
+    }
+
+    fn name(&self) -> String {
+        "uninitialized_use_checker".to_string()
+    }
+}
+
+// ====================================================================
+// Formatting functionality for initialized state annotation.
+
+/// Format the initialized state annotation for a given function target.
+pub fn format_initialized_state_annotation(
+    target: &FunctionTarget,
+    code_offset: CodeOffset,
+) -> Option<String> {
+    let InitializedStateAnnotation(map) = target
+        .get_annotations()
+        .get::<InitializedStateAnnotation>()?;
+    let (before, after) = map.get(&code_offset)?;
+    let mut s = String::new();
+    s.push_str("before: ");
+    s.push_str(&format_initialized_state(before, target));
+    s.push_str(", after: ");
+    s.push_str(&format_initialized_state(after, target));
+    Some(s)
+}
+
+/// Format a vector of `locals`.
```

Comment:
It it just a private helper used in other formatters in this file, to reduce code duplication.

Because these are used for annotation formatting, which can be specific to the analyses, I am not sure if this specific style of formatting is more generally useful and thus belongs as a public method in the `FunctionTarget` interface.